wicket/src/state/rack.rs

```diff
@@ -0,0 +1,164 @@
+    left_column: bool,
+}
+
+#[allow(clippy::new_without_default)]
+impl RackState {
+    pub fn new() -> RackState {
+        RackState {
+            log: None,
+            selected: ComponentId::Sled(0),
+            knight_rider_mode: None,
+
+            // Default to the left column, where sled 0 lives
+            left_column: true,
+        }
+    }
+
+    pub fn toggle_knight_rider_mode(&mut self) {
+        if self.knight_rider_mode.is_none() {
+            self.knight_rider_mode = Some(KnightRiderMode::default());
+        } else {
+            self.knight_rider_mode = None;
+        }
+    }
+
+    pub fn up(&mut self) {
+        self.selected = match self.selected {
+            ComponentId::Sled(i) if i == 14 || i == 15 => {
+                ComponentId::Switch(0)
+            }
+            ComponentId::Sled(i) => ComponentId::Sled((i + 2) % 32),
+            ComponentId::Switch(0) => ComponentId::Psc(0),
+            ComponentId::Switch(1) => {
+                if self.left_column {
+                    ComponentId::Sled(16)
+                } else {
+                    ComponentId::Sled(17)
+                }
+            }
+            ComponentId::Switch(_) => unreachable!(),
+            ComponentId::Psc(0) => ComponentId::Psc(1),
+            ComponentId::Psc(1) => ComponentId::Switch(1),
+            ComponentId::Psc(_) => unreachable!(),
+        };
+    }
+
+    pub fn down(&mut self) {
+        self.selected = match self.selected {
+            ComponentId::Sled(i) if i == 16 || i == 17 => {
+                ComponentId::Switch(1)
+            }
+            ComponentId::Sled(i) => ComponentId::Sled((30 + i) % 32),
+            ComponentId::Switch(1) => ComponentId::Psc(1),
+            ComponentId::Switch(0) => {
+                if self.left_column {
+                    ComponentId::Sled(14)
+                } else {
+                    ComponentId::Sled(15)
+                }
+            }
+            ComponentId::Switch(_) => unreachable!(),
+            ComponentId::Psc(0) => ComponentId::Switch(0),
+            ComponentId::Psc(1) => ComponentId::Psc(0),
+            ComponentId::Psc(_) => unreachable!(),
+        };
+    }
+
+    pub fn left_or_right(&mut self) {
+        match self.selected {
+            ComponentId::Sled(i) => {
+                if self.left_column {
+                    self.left_column = false;
+                    self.selected = ComponentId::Sled(i + 1);
+                } else {
+                    self.left_column = true;
+                    self.selected = ComponentId::Sled(i - 1);
+                }
+            }
+            _ => (),
+        }
+    }
+
+    pub fn next(&mut self) {
+        self.selected = match self.selected {
+            ComponentId::Sled(i)
+                if ((0..=14).contains(&i) || (16..=31).contains(&i)) =>
+            {
+                ComponentId::Sled((i + 1) % 32)
+            }
+            ComponentId::Sled(15) => ComponentId::Switch(0),
```

Comment:
If you put this one first, I think you could remove the long `if...` on the other `Sled` case entirely. (Same comment on `prev()`)

Reply:
Wow, great catch! Fixed in [fd4ea2d](https://github.com/oxidecomputer/omicron/pull/2440/commits/fd4ea2d1c6e5f6a298b987e48eeefb5ee6bbeaf0)